sparksteps/__main__.py

```diff
@@ -170,6 +170,15 @@ def determine_prices(args, ec2, pricing_client):
     if not any([x in args for x in pricing_properties]):
         return args
 
+    availability_zone = None
+    if 'ec2_subnet_id' in args and args['ec2_subnet_id']:
+        # We need to determine the AZ associated with the provided EC2 subnet ID
+        # in order to look up spot prices in the correct region.
+        subnet_id = args['ec2_subnet_id']
```

Comment:
Appears that `args` is provided as a `dict` so `get()` would work well here:
```suggestion
    subnet_id = args.get('ec2_subnet_id')
    if subnet_id:
        # We need to determine the AZ associated with the provided EC2 subnet ID
        # in order to look up spot prices in the correct region.
```